File: services/go/api/utils/aws.go
package utils

import (
	"bytes"
	"os"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/credentials"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/s3/s3manager"
	"github.com/aws/aws-sdk-go/service/ses"
)

func SESSession() (*session.Session, error) {
	return session.NewSession(&aws.Config{
		Region:      aws.String(os.Getenv("AWS_REGION")),
		Credentials: credentials.NewStaticCredentials(os.Getenv("AWS_ACCESS_KEY"), os.Getenv("AES_SECRET_KEY"), ""),
	})
}

func SendEmail(template string, to string, data string) (ok bool, err error) {
	session := session.Must(SESSession())
	svc := ses.New(session)

	email := &ses.SendTemplatedEmailInput{
		Source:   aws.String(os.Getenv("AWS_SES_FROM")),
		Template: &template,
		Destination: &ses.Destination{
			ToAddresses: []*string{aws.String(to)},
		},
		TemplateData: &data,
	}

	_, emailErr := svc.SendTemplatedEmail((email))

	if emailErr != nil {
		println("[EMAIL ERROR]", emailErr.Error())

		return false, emailErr
	}

	return true, nil
}

func SaveImage(imageName string, buf []byte) (ok bool, err error) {
	session := session.Must(SESSession())
	uploader := s3manager.NewUploader(session)

	result, err := uploader.Upload(&s3manager.UploadInput{
		Bucket: aws.String("imperial"),
		Key:    aws.String(imageName + ".png"),
		Body:   bytes.NewReader(buf),
	})

	if err != nil && result == nil {
		return false, err
	}

	return true, nil
}

File: services/node/api/src/utils/aws.ts
import { S3 as S3Client, SES as SESClient } from "aws-sdk";
import { env } from "./env";

const s3 = new S3Client({
  region: env.AWS_REGION,

  accessKeyId: env.AWS_ACCESS_KEY,
  secretAccessKey: env.AWS_SECRET_KEY,
});

const ses = new SESClient({
  region: env.AWS_REGION,
  credentials: {
    accessKeyId: env.AWS_ACCESS_KEY,
    secretAccessKey: env.AWS_SECRET_KEY,
  },
});

class S3 {
  public static async uploadFile(
    key: string,
    body: string,
    contentType: string
  ) {
    const uploadRequest = await s3
      .upload({
        Bucket: "imperial",
        Key: key,
        Body: body,
        ContentType: contentType,
      })
      .promise();

    console.log(uploadRequest.Location);

    return uploadRequest;
  }
}

export { S3 };
